app/models/manageiq/providers/redhat/inventory/persister/definitions/infra_collections.rb

```diff
@@ -24,7 +24,9 @@ def initialize_infra_inventory_collections
   # --- IC groups definitions ---
 
   def add_clusters_group
-    add_collection(infra, :ems_clusters)
+    add_collection(infra, :ems_clusters) do |builder|
+      builder.add_properties(:model_class => ::EmsCluster)
```

Comment:
@borod108  and with the nested definition ^, lets try

```:model_class => ManageIQ::Providers::Redhat::InfraManager::EmsCluster```

and then verify it stores the right :type into the DB